packages/vite/src/plugin.ts

```diff
@@ -0,0 +1,57 @@
+import { resolver } from './resolver.js';
+import { hbs } from './hbs.js';
+import { scripts } from './scripts.js';
+import { templateTag } from './template-tag.js';
+import { optimizeDeps } from './optimize-deps.js';
+import { compatPrebuild } from './build.js';
+import { assets } from './assets.js';
+import { contentFor } from './content-for.js';
+import { babel } from '@rollup/plugin-babel';
+import { type ConfigEnv, type UserConfig } from 'vite';
+
+export const extensions = ['.mjs', '.gjs', '.js', '.mts', '.gts', '.ts', '.hbs', '.json'];
+
+export default function ember() {
+  return [
+    {
+      name: 'vite-plugin-ember',
+      enforce: 'pre',
+      config(_config: UserConfig, env: ConfigEnv) {
+        return {
+          resolve: {
+            extensions,
+          },
+          optimizeDeps: optimizeDeps(),
+          server: {
+            port: 4200,
+          },
+          build: {
+            outDir: 'dist',
+            rollupOptions: {
+              input: {
+                main: 'index.html',
+                ...(shouldBuildTests(env.mode) ? { tests: 'tests/index.html' } : undefined),
+              },
+            },
+          },
+        };
+      },
+    },
+    hbs(),
+    templateTag(),
+    scripts(),
+    resolver(),
+    compatPrebuild(),
+    assets(),
+    contentFor(),
+
+    babel({
```

Comment:
I liked your earlier suggestion of having babel present in the userland vite config.

folks may want to change this and it isn't "our" plugin

Reply:
babel would be configured with the babel config file, not in the vite config. 

See react and vue vite plugins, they also setup babel but it's ultimately configured by the user with a babel config file.
https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react/src/index.ts
https://github.com/vitejs/vite-plugin-vue/blob/599f81369816aca918314e86f5b0a6e32367bb38/packages/plugin-vue-jsx/src/index.ts#L4

Reply:
I'll leave this one open to see if anyone else has an opinion. I would vote for keeping this internal though, as users should only need to worry about `babel.config.js`, not how it actually is run on the files.